src/descriptors.js

```diff
@@ -124,28 +124,13 @@ export function cloneDescriptor(d: void | PropertyDescriptor): void | PropertyDe
 
 // does not check if the contents of value properties are the same
 export function equalDescriptors(d1: PropertyDescriptor, d2: PropertyDescriptor): boolean {
-  if (d1.hasOwnProperty("writable")) {
-    if (!d2.hasOwnProperty("writable")) return false;
-    if (d1.writable !== d2.writable) return false;
-  }
-  if (d1.hasOwnProperty("enumerable")) {
-    if (!d2.hasOwnProperty("enumerable")) return false;
-    if (d1.enumerable !== d2.enumerable) return false;
-  }
-  if (d1.hasOwnProperty("configurable")) {
-    if (!d2.hasOwnProperty("configurable")) return false;
-    if (d1.configurable !== d2.configurable) return false;
-  }
-  if (d1.hasOwnProperty("value")) {
-    if (!d2.hasOwnProperty("value")) return false;
-  }
-  if (d1.hasOwnProperty("get")) {
-    if (!d2.hasOwnProperty("get")) return false;
-    if (d1.get !== d2.get) return false;
-  }
-  if (d1.hasOwnProperty("set")) {
-    if (!d2.hasOwnProperty("set")) return false;
-    if (d1.set !== d2.set) return false;
-  }
+  if (d1.writable !== d2.writable) return false;
+  if (d1.enumerable !== d2.enumerable) return false;
+  if (d1.configurable !== d2.configurable) return false;
+  if (d1.value !== undefined) {
```

Comment:
There is some asymmetry here, causing equalDescriptors not to be commutative when one `value` is `undefined` and the other one isn't. That issue already existed in the original code. I doubt that it's intentional? Maybe it gets rescued by the other invariant that the existance of `value` and `get`/`set` are mutually exclusive.

Anyway, for clarity, I'd suggest rewriting this case to
```js
if ((d1.value === undefined) !== (d2.value === undefined)) return false;
```

Reply:
There are three types of descriptors at the spec level. Generic, Data and Accessors. Lack of value and get/set means it's generic. So if one has a value but the other doesn't, it means that they're not even in the same category of descriptors.

Undefined value doesn't mean `empty` btw. That's an empty value.

This helper function, for reasons that preceded me, doesn't compare whether the values themselves are equivalent.

Basically, this function is only meant to determine if this needs to use joined Descriptor or if it's enough to use a joined Value.

In theory, accessors doesn't need to use a joined Descriptor neither. We can just use a joined value of the getter and a joined value of the setter.